### ISSUE TITLE
Type resolution error for jasmine test function `it` if build output folder is not empty

### ISSUE THREAD

**Issue by raphaelmeyer** [angular-eslint/angular-eslint#1814]
I encountered a strange issue causing a lint error, because of failed type resolution. I could not find a culprit or workaround while messing around with tsconfig or eslint configuration. But at least I could reproduce the issue with a few setup steps.

**Description and reproduction of the issue**

Create an angular workspace, add angular-eslint and generate an application:

```sh
npx -p @angular/cli@17 ng new --create-application=false --strict example
cd example
npx ng add @angular-eslint/schematics
npx ng generate application --style=scss --ssr=false poiuqwer
```

In file `example/.eslintrc.json` in the `overrides` section for `"files": ["*.ts"]` add
  - option  `"parserOptions": { "project": true }`
  - rule `"@typescript-eslint/no-unsafe-call": "error"`

Running command `npx ng build && npx ng lint` will report errors for the added rule, because it fails to resolve the type of the test functions `it`.

```sh
Linting "poiuqwer"...

/home/user/Sandbox/eslint-error/example/projects/poiuqwer/src/app/app.component.spec.ts
  11:3  error  Unsafe call of an `any` typed value  @typescript-eslint/no-unsafe-call
  17:3  error  Unsafe call of an `any` typed value  @typescript-eslint/no-unsafe-call
  23:3  error  Unsafe call of an `any` typed value  @typescript-eslint/no-unsafe-call

✖ 3 problems (3 errors, 0 warnings)

Lint errors found in the listed files.
```

But after deleting the build output, it works as expected: `rm -r ./dist/ && npx ng lint`.

```sh
Linting "poiuqwer"...

All files pass linting.
```

The issue persists if jasmine/karma is replaced with jest. In such a setup I observed that eslint failed at jest alias `it`, but had no problem with alias `test` (both aliasing the same type).

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/builder`          | `17.4.0` |
| `ESLint`                           | `8.57.0` |
| `node`                             | `20.11.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:

Angular CLI: 17.3.7
Node: 20.11.1
Package Manager: npm 10.2.4
OS: linux x64

Angular: 17.3.8
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1703.7
@angular-devkit/build-angular   17.3.7
@angular-devkit/core            17.3.7
@angular-devkit/schematics      17.3.7
@angular/cli                    17.3.7
@schematics/angular             17.3.7
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.5
```

**Comment by StefanNedelchev** [comment]
I encountered it too and somehow it disappeared without me realizing how it happened. Thanks for providing an explanation about it and I hope it would help the team to fix the issue.

**Comment by JamesHenry** [comment]
I'm sorry for the delay on this one @raphaelmeyer, however none of the moving parts here apply to angular-eslint, so I'm going to close this one.

If you continue experience issues with the behavior of `project: true` and what files it is picking up on the latest versions of typescript-eslint please report it on https://github.com/typescript-eslint/typescript-eslint with a minimal reproduction.

Many thanks!
